src/services/routeMatchRules.spec.ts

```diff
@@ -168,6 +170,26 @@ describe('routeQueryMatches', () => {
     expect(response).toBe(true)
   })
 
+  test.each([
+    ['?default'],
+    ['?default='],
```

Comment:
What is the param value for these two cases? 

Also should there be a test case where the param is completely missing?

Reply:
I would expect the default value `abc` to be the value. Worth testing though

I also would like to test if the param is completely missing. Let's leave this comment unresolved until I do these 2 thigns

Reply:
I added cases for "" and "?", I also verified that the default value WILL be present in `route.params` for these cases will be populated with default value